fingerprint/admin.py

```diff
@@ -0,0 +1,268 @@
+from datetime import timedelta
+from functools import wraps
+from itertools import chain
+from typing import Callable, Iterator
+
+from django.contrib import admin
+from django.db.models import Count, Max, Prefetch, Q
+from django.urls import reverse
+from django.utils.html import format_html
+from django.utils.timezone import now
+
+from .models import BrowserFingerprint, RequestFingerprint, UserFingerprint, UserSession
+
+try:
+    from itertools import pairwise
+except ImportError:
+    def pairwise(items):
+        return zip(items, items[1:])
+
+
+class html_objects_list:
+    def __init__(self, format_string: str):
+        self.format_string = format_string
+
+    def __call__(self, fn: Callable) -> Callable:
+
+        @wraps(fn)
+        def wrapped(*args, **kwargs):
+            results = list(fn(*args, **kwargs))
+            return format_html(
+                '<br>'.join(self.format_string for _ in results),
+                *chain.from_iterable(results)
+            )
+
+        return wrapped
+
+
+@admin.register(UserSession)
+class UserSessionAdmin(admin.ModelAdmin):
+    list_display = 'session_key', 'user', 'browser_fingerprints', 'request_fingerprints', 'created',
+    list_filter = 'created',
+    search_fields = 'user__username', 'session_key',
+    ordering = '-created',
+
+    def get_queryset(self, request):
+        return (
+            super().get_queryset(request)
+            .select_related('user')
+            .prefetch_related('browserfingerprints', 'requestfingerprints')
+        )
+
+    @html_objects_list('{} &nbsp;&nbsp; <a href="{}"><tt>{}</tt></a>')
```

Comment:
nitpick: `tt` tag is deprecated, please consider using `code` or something different
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tt